contracts/DelFiPrice.sol

```diff
@@ -170,20 +175,66 @@ contract DelFiPrice is OpenOracleView {
      * @return true, this contract is meant to be used by Compound v2 PriceOracle interface.
      */
     function isPriceOracle() external pure returns (bool) {
-       return true;
+        return true;
+    }
+
+    // fetchs price in eth from proxy and converts to usd price using anchor usdc price
+    // anchor usdc price has 30 decimals, and anchor general price has 18 decimals,
+    // so multiplying by 1e18 and by 1e30 yields 1e6
+    function getAnchorPrice(string memory symbol, uint256 usdcPrice) public  returns (uint64) {
+        address tokenAddress = getCTokenAddress(symbol);
+
+        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
+            // hard code to 1 dollar
+            return 1e6;
+        }
+
+        uint priceInEth = anchor.getUnderlyingPrice(tokenAddress);
+        uint additionalScale;
+        if ( tokenAddress == cWbtcAddress ){
```

Comment:
Maybe we should have some kind of mapping that is "decimals" so we can just have a clean way of getting this? Here's a thought, we have an immutable `TokenData` struct that's returns from Toni's case statement below and passed around? It could have Oracle symbol, reporter list, decimals, etc, and just be hard-coded in the constructor. This could make life :oh so nice:

Reply:
I really wanted immutable structs to be a thing too, but they're not. (we could probably fake it better than this though )

Reply:
but setting in constants or a function, yes will take a look and see if it looks nicer

Reply:
oh you mean like an in memory struct? does that work? that would be next best thing to real `immutable` struct